gr-digital/lib/constellation.cc

```diff
@@ -239,16 +245,22 @@ void constellation::gen_soft_dec_lut(int precision, float npwr)
 {
     d_soft_dec_lut.clear();
     d_lut_scale = powf(2.0f, static_cast<float>(precision));
+    //check if noise power is overriden. use global if not
+    if (npwr != d_npwr && npwr != 1.0){d_npwr = npwr;}
+    else {npwr = d_npwr;}
 
     // We know we've normalized the constellation, so the min/max
     // dimensions in either direction are scaled to +/-1.
     float maxd = 1.0f;
+    //the above comment isn't really true, but LUT is funky so we'll scale our points inside the +-1 LUT
+    //we will multiply inputs to obtain LLR's for points with the scale ptscale instead of +-1
+    float ptscale = 2.0f*d_maxamp;
```

Comment:
I think you've mentioned that one of the problems hurting the BER performance of the LDPC decoder was that the range of `x` and `y` for which the LUT was built was not appropriate, so you've increased it from `[-1, 1]` to `[-2*d_maxamp, 2*d_maxamp]`.

This seems reasonable. `[-1, 1]` is simply too small, specially for constellations whose outermost points have amplitudes much larger than one.

I wonder, is there a better choice than `[-2*d_maxamp, 2*d_maxamp]`? Maybe we can make the range slightly smaller than this to gain better resolution without getting too much clipping problems (I assume that the LUT lookup function just clips to the edge value if the `x` or `y` coordinates are outside of the LUT range). Since we're constructing this for an expected noise power, we can use the normal distribution CDF to help us. For example, if we set
```
float ptscale = d_maxamp + 2.0 * sqrtf(npwr);
```
then I think that only less than 2.2% of the LUT lookups will be outside the range. This follows because only 2.2% of the mass of the normal distribution falls above 2*sigma. So to fall outside of `ptscale`, we would need that the symbol corresponds to one of the outermost constellation points that have amplitude `d_maxamp` and that we are so unlucky that the complex noise points towards the "outside" direction and has a large amplitude (here I'm doing some handwaving/approximations, the amplitude of complex Gaussian noise actually has a Rayleigh distribution).

My choice of using 2*sigma in the above is somewhat arbitrary. We can increase it (3*sigma) or reduce it (1*sigma) depending on how bad we believe (or check) that LUT clipping is.

I'm not sure how my idea compares with your choice of `float ptscale = 2.0*d_maxamp`. Mine seems more inspired in how to avoid clipping, and will make the box larger when there is a lot of noise, and smaller when it can afford to reduce it without clipping because there is less noise. But yours seems a more simple choice. I'm happy to keep either your proposal or mine, since your proposal also probably does quite well in practice.

Reply:
@daniestevez So the idea of selecting N standard deviations to pad with was the first thought I had too. However, the 2x padding ended up being simpler as you guessed. Mainly this came down to debugging issues, so perhaps it can change now that things are working properly. On the other hand, there are a few advantages to 2x padding. Firstly, IIRC some standard like DVB adopted this technique for their soft LUT. Second, it makes determining the resolution loss very simple, the constellation is X-1 bits dense no matter the noise level. This somewhat simplifies the analysis and determination of necessary LUT resolution. Lastly, it almost guarantees no clipping under many cases. 
But again, I'm not too attached here, it just made sense when writing the code.

Reply:
I'm thinking that you're right in the sense that simplicity in the 2x padding is quite desirable. Users concerned with quantization losses can simply set a larger `precision` to build a larger LUT. So I'm happy to keep the 2x padding.